uw-research-computing/singularity-htc.md

```diff
@@ -0,0 +1,167 @@
+    apt-get install -y \  # command to install building tools we may need
+            build-essential \
+            cmake \
+            g++ \
+            r-base-dev
+
+    R -e "install.packages('cowsay', dependencies=TRUE, repos='http://cran.rstudio.com/')" 
+```
+The `R -e "install.packages('cowsay', dependencies=TRUE, repos='http://cran.rstudio.com/')" ` line tells our image to start `R` and to use the common `R` fucnction `install.packages` to install `cowsay`. 
+
+See the Apptainer documentation for a full reference on how to specify build specs. Note that the `%runscript` section is ignored when the container is executed on the High Throughput system.
+
+The final `image.def` looks like:
+
+```
+Bootstrap: docker
+From: rocker/geospatial:4.2.2
+
+%post
+    apt-get update -y
+    apt-get install -y \
+            build-essential \
+            cmake \
+            g++ \
+            r-base-dev
+
+    R -e "install.packages('cowsay', dependencies=TRUE, repos='http://cran.rstudio.com/')"
+```
+
+## 4. Build your Apptainer Container
+
+Once your build spec is ready, you can "build" the container. The process of building an apptainer image will convert our `.def` file to a `.sif` Apptainer/Singularity image that we will use for future job submissions. To build our container, run this command:
+
+```
+$ apptainer build my-container.sif image.def
+```
+
+If you list the contents of your scratch directory by running `ls`, you should now see your executable Apptainer/Singularity file `my-container.sif`. 
+
+Once the image is built, it is important to test it to make sure you have all software, packages, and libraries installed correctly. 
+
+## 5. Testing Your Container
+
+Just like it is important to test your codes and jobs at a small scale, you should make sure that your container is working correctly. To do this, we will start an "interactive Apptainer session" with the Apptainer/Singularity "shell" mode. The recommended command line, similar to how containers are started for jobs, is:
+
+```
+THIS CODE NEEDS TESTING (LIKELY REMOVE /SRV, ETC)
+apptainer shell my-container.sif \
+            --home $PWD:/srv \
+            --pwd /srv \
+           #--bind /cvmfs \
+            --scratch /var/tmp \
+            --scratch /tmp \
+            --contain --ipc --pid \
+``` 
+
+This will give you an interactive shell using your container, with your current working directory mounted under /srv. Your prompt should change to: 
+
+```
+Apptainer my-container.sif:~>
+```
+
+You can explore the container and test your code in this mode. Once you are down exploring, exit the container by running `exit` or with `CTRL+D`.
+ 
+
+# 6. Exit your HTCondor Interactive Job
+
+Now that we have prepared our Apptainer image and tested it, we are ready to exit our interactive session. When we do, our build file will return to our `/home` directory on the submit server.  To exit, run:
+
+```
+[alice@build]$ exit
+```
+
+## 7. Use an Apptainer Container in HTC Jobs
```

Comment:
I would actually make this first in the guide. I'll submit a quick PR against the Docker guide to show what I mean.